plot-lib/plot/private/no-gui/plot2d.rkt

```diff
@@ -131,6 +133,78 @@
                    #:title title #:x-label x-label #:y-label y-label #:legend-anchor legend-anchor)))
       width height))
 
+(:: legend-pict
+    (->* [(Treeof (U renderer2d nonrenderer))]
+         [#:x-min (U Real #f) #:x-max (U Real #f)
+          #:y-min (U Real #f) #:y-max (U Real #f)]
+         Pict))
+(define (legend-pict renderer-tree
+                     #:x-min [x-min #f] #:x-max [x-max #f]
+                     #:y-min [y-min #f] #:y-max [y-max #f])
+  (define fail/kw (make-raise-keyword-error 'plot/dc))
+  (cond
+    [(and x-min (not (rational? x-min)))  (fail/kw "#f or rational" '#:x-min x-min)]
+    [(and x-max (not (rational? x-max)))  (fail/kw "#f or rational" '#:x-max x-max)]
+    [(and y-min (not (rational? y-min)))  (fail/kw "#f or rational" '#:y-min y-min)]
+    [(and y-max (not (rational? y-max)))  (fail/kw "#f or rational" '#:y-max y-max)]
+    [else
+     (define saved-values (plot-parameters))
+  
+     (define renderer-list (get-renderer-list renderer-tree))
+     (define bounds-rect (get-bounds-rect renderer-list x-min x-max y-min y-max))
+     (define legend (get-legend-entry-list renderer-list bounds-rect))
+     
+     (define-values (width height)
+       (parameterize/group ([plot-parameters saved-values])
```

Comment:
I don't think we need `parameterize/group` here -- this code runs in the same thread that captured the parameters.